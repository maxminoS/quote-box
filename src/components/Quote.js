import React from "react";

export default props => (
  <>
    {props.quote
     ? <>
         <h3>{props.quote.quote}</h3>
         <p>- {props.quote.author}</p>
       </>
     : ""}
  </>
);
