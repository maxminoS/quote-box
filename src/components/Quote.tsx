import React from "react";
import { Typography } from "@material-ui/core";
import { FormatQuote } from "@material-ui/icons";

type Props = {
  quote: {
    quote: string,
    author: string,
  }
}

export const Quote = (props: Props) => (
  <>
    {props.quote
     ? <>
         <FormatQuote fontSize="large" />
         <Typography variant="h6">
           {props.quote.quote}
         </Typography>
         <Typography variant="subtitle1"
                     align="right"
         >
           - {props.quote.author}
         </Typography>
       </>
     : ""}
  </>
);
