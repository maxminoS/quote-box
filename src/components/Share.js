import React from "react";
import { IconButton } from '@material-ui/core';
import { Twitter } from '@material-ui/icons';

export default props => (
  <>
    {props.quote
     ? <IconButton onClick={() => window.open("https://twitter.com/intent/tweet?text=" + encodeURIComponent("\"" + props.quote.quote + "\"\n- " + props.quote.author))
                           }
       >
         <Twitter color="primary" fontSize="large" />
       </IconButton>
     : ""}
  </>
);
