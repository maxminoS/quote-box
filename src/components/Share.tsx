import React from "react";
import { IconButton } from '@material-ui/core';
import { Twitter } from '@material-ui/icons';

type Props = {
  quote: {
    quote: string,
    author: string,
  }
}

export default (props: Props) => (
  <>
    {props.quote
     ? <IconButton onClick={() => window.open("https://twitter.com/intent/tweet?text=" + encodeURIComponent("\"" + props.quote.quote + "\"\n- " + props.quote.author))
                           }
       >
         <Twitter color="primary" fontSize="large" />
       </IconButton>
     : ""}
  </>
);
