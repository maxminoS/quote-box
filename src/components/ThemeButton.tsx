import React from "react";
import { Grid, Tooltip, IconButton } from '@material-ui/core';
import { Brightness3, WbSunny } from "@material-ui/icons";

type Props = {
  darkMode: boolean,
  setDarkMode: React.Dispatch<React.SetStateAction<boolean>>
}

export const ThemeButton = (props: Props) => (
  <Grid item
        style={{
          position: "absolute",
          bottom: 8,
          right: 8
        }}
  >
    <Tooltip title={ props.darkMode ? "Dark Mode" : "Light Mode"} placement="left" arrow>
      <IconButton onClick={() => props.setDarkMode(!props.darkMode)}>
        { props.darkMode ? <Brightness3 /> : <WbSunny /> }
      </IconButton>
    </Tooltip>
  </Grid>
);
