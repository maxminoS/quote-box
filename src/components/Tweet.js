import React from "react";

export default props => (
  <>
    {props.quote
     ? <a href={"https://twitter.com/intent/tweet?text="
                + escape("\"" + props.quote.quote + "\"\n- " + props.quote.author)
               }
         target="_blank"
       >
         Tweet
       </a>
     : ""}
  </>
);
