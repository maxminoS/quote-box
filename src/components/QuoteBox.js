import React, { useState, useEffect } from "react";
import { Button } from "@material-ui/core";
import { Card, CardContent, CardActions } from "@material-ui/core";

import Quote from "components/Quote";
import Share from "components/Share";
import QuoteApi from "api/QuoteApi";

export default () => {
  const [quotes, setQuotes] = useState([]);
  const [quoteNumber, setQuoteNumber] = useState(0);

  useEffect(async () => {
    QuoteApi.getQuote()
      .then(quotes => {
        setQuotes(quotes);
        setQuoteNumber(Math.floor(Math.random() * quotes.length));
      });
  }, []);

  const handleClick = () => {
    setQuoteNumber(Math.floor(Math.random() * quotes.length));
  };

  return (
    <Card raised="true">
      <CardContent>
        <Quote
          quote={quotes[quoteNumber]}
        />
      </CardContent>
      <CardActions>
        <Share
          quote={quotes[quoteNumber]}
        />
        <Button variant="contained" color="secondary" style={{marginLeft: 'auto'}} onClick={handleClick}>
          GET
        </Button>
      </CardActions>
    </Card>
  );
}
