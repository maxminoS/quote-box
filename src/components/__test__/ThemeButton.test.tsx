import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';

import { ThemeButton }from 'components/ThemeButton';

describe('ThemeButton', () => {
  it('can click the icon button', () => {
    render(<ThemeButton darkMode={false} setDarkMode={() =>  {}} />);
    userEvent.click(screen.getByRole('button'));
  });

  it('renders light ThemeButton component as expected', () => {
    const themeButtonLight = renderer.create(<ThemeButton darkMode={false} setDarkMode={() =>  {}} />).toJSON();
    expect(themeButtonLight).toMatchSnapshot();
  });

  it('renders dark ThemeButton component as expected', () => {
    const themeButtonDark = renderer.create(<ThemeButton darkMode={true} setDarkMode={() =>  {}} />).toJSON();
    expect(themeButtonDark).toMatchSnapshot();
  });
})
