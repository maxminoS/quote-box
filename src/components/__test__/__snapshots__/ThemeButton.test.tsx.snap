// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeButton renders dark ThemeButton component as expected 1`] = `
<div
  className="MuiGrid-root MuiGrid-item"
  style={
    Object {
      "bottom": 8,
      "position": "absolute",
      "right": 8,
    }
  }
>
  <button
    aria-describedby={null}
    className="MuiButtonBase-root MuiIconButton-root"
    data-testid="theme-button"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    title="Dark Mode"
    type="button"
  >
    <span
      className="MuiIconButton-label"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M9 2c-1.05 0-2.05.16-3 .46 4.06 1.27 7 5.06 7 9.54 0 4.48-2.94 8.27-7 9.54.95.3 1.95.46 3 .46 5.52 0 10-4.48 10-10S14.52 2 9 2z"
        />
      </svg>
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>
</div>
`;

exports[`ThemeButton renders light ThemeButton component as expected 1`] = `
<div
  className="MuiGrid-root MuiGrid-item"
  style={
    Object {
      "bottom": 8,
      "position": "absolute",
      "right": 8,
    }
  }
>
  <button
    aria-describedby={null}
    className="MuiButtonBase-root MuiIconButton-root"
    data-testid="theme-button"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    title="Light Mode"
    type="button"
  >
    <span
      className="MuiIconButton-label"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M6.76 4.84l-1.8-1.79-1.41 1.41 1.79 1.79 1.42-1.41zM4 10.5H1v2h3v-2zm9-9.95h-2V3.5h2V.55zm7.45 3.91l-1.41-1.41-1.79 1.79 1.41 1.41 1.79-1.79zm-3.21 13.7l1.79 1.8 1.41-1.41-1.8-1.79-1.4 1.4zM20 10.5v2h3v-2h-3zm-8-5c-3.31 0-6 2.69-6 6s2.69 6 6 6 6-2.69 6-6-2.69-6-6-6zm-1 16.95h2V19.5h-2v2.95zm-7.45-3.91l1.41 1.41 1.79-1.8-1.41-1.41-1.79 1.8z"
        />
      </svg>
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>
</div>
`;
